### PR TITLE
 Fix periodicity condition in YTPoint.

### LINKED ISSUE #1711: YTPoint doesn't work for negative coordinates
Issue was reported via email by @mogrodnik. Looks like `YTPoint` doesn't like negative coordinates:

```
>>> import yt
>>> yt.get_yt_version()
'b38bdb8728a5'
>>> yt.__version__
'3.5.dev0'
>>> ds = yt.load('sedov/sedov_tst_0004.h5')  # wget http://yt-project.org/data/sedov_tst_0004.tar.gz yt : [WARNING  ] 2018-03-04 14:18:18,617 'field_units' was overridden by 'dataset_units/density'
yt : [WARNING  ] 2018-03-04 14:18:18,619 'field_units' was overridden by 'dataset_units/level'
yt : [WARNING  ] 2018-03-04 14:18:18,619 'field_units' was overridden by 'dataset_units/mag_field_x'
yt : [WARNING  ] 2018-03-04 14:18:18,620 'field_units' was overridden by 'dataset_units/mag_field_y'
yt : [WARNING  ] 2018-03-04 14:18:18,621 'field_units' was overridden by 'dataset_units/mag_field_z'
yt : [WARNING  ] 2018-03-04 14:18:18,623 'field_units' was overridden by 'dataset_units/specific_energy'
yt : [WARNING  ] 2018-03-04 14:18:18,626 'field_units' was overridden by 'dataset_units/velocity_x'
yt : [WARNING  ] 2018-03-04 14:18:18,627 'field_units' was overridden by 'dataset_units/velocity_y'
yt : [WARNING  ] 2018-03-04 14:18:18,627 'field_units' was overridden by 'dataset_units/velocity_z'
yt : [INFO     ] 2018-03-04 14:18:18,643 Parameters: current_time              = [ 0.00020541]
yt : [INFO     ] 2018-03-04 14:18:18,643 Parameters: domain_dimensions         = [64 64 64]
yt : [INFO     ] 2018-03-04 14:18:18,643 Parameters: domain_left_edge          = [-1. -1. -1.]
yt : [INFO     ] 2018-03-04 14:18:18,644 Parameters: domain_right_edge         = [ 1.  1.  1.]
yt : [INFO     ] 2018-03-04 14:18:18,644 Parameters: cosmological_simulation   = 0.0
>>> ad = ds.all_data()
>>> ad.min('density')
0.30130037665367126 dimensionless**(-2)
>>> ad.argmin('density')
(-0.08203125 code_length, -0.14453125 code_length, -0.14453125 code_length)
>>> pt = ds.r[ad.argmin('density')]
>>> pt
YTPoint (sedov_tst_0004): , p=[-0.08203125 -0.14453125 -0.14453125] cm
>>> pt['density']
YTArray([], dtype=float64) dimensionless**(-2)
```

### PR DESCRIPTION
**Pull request by Xarthisius**
## PR Summary

Previous periodicity condition in `YTPoint` constructor assumed that the domain starts with 0.

### CODE REVIEW

File: yt/utilities/tests/test_selectors.py [hunk 1]
```diff
@@ -10,21 +10,22 @@ def setup():
...
 def test_point_selector():
     # generate fake amr data
-    ds = fake_random_ds(64, nprocs=51)
+    bbox = np.array([[-1.0, 1.0], [-1.0, 1.0], [-1.0, 1.0]])
+    ds = fake_random_ds(16, nprocs=7, bbox=bbox)
     assert(all(ds.periodicity))
 
     dd = ds.all_data()
     positions = np.array([dd[ax] for ax in 'xyz'])
     delta = 0.5*np.array([dd['d'+ax] for ax in 'xyz'])
```
Comment (ngoldbaum) [inline yt/utilities/tests/test_selectors.py]: nit: it's a little clearer if you do the transpose here

**Comment by ngoldbaum** [comment]
Great catch @mogrodnik! And thank you for the quick PR with tests @xarthisius. This LGTM.

**Comment by jzuhone** [comment]
@Xarthisius It's not merely an issue of periodicity as there is no reason a non-periodic dataset wouldn't have `DLE < 0.0` (plenty of my FLASH datasets do). It looks like this would work with that, but can you verify?

**Comment by ngoldbaum** [comment]
@jzuhone yup, see the modifications @Xarthisius made to the point selector test.

**Comment by Xarthisius** [comment]
@jzuhone yup, in the initial bug report, we were selecting a point **within** the existing domain (just < 0.0). It was periodicity condition that was messing it up.

**Comment by jzuhone** [comment]
@Xarthisius @ngoldbaum ok, thanks, it looked like that, but I wanted to double-check. LGTM.

**Pull request merged**